external-crates/move/tools/move-cli/src/base/new.rs

```diff
@@ -48,6 +52,19 @@ impl New {
     ) -> anyhow::Result<()> {
         // TODO warn on build config flags
         let Self { name } = self;
+
+        let valid_identifier_re = match Regex::new(r"^[A-Za-z][A-Za-z\_\-]*$") {
+            Ok(re) => re,
+            Err(_) => {
+                return Err(anyhow!(
+                    "Cannot build the regex needed to validate package naming"
+                ));
+            }
+        };
+
+        if !valid_identifier_re.is_match(name.as_str()) {
```

Comment:
Is name a `String`? I feel like it is more canonical to do 
```suggestion
        if !valid_identifier_re.is_match(&name) {
```